lib/TPP/VNNIUtils.cpp

```diff
@@ -0,0 +1,22 @@
+//===- VNNIUtils.cpp ---------------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "TPP/VNNIUtils.h"
+
+namespace mlir {
+namespace vnni {
+namespace utils {
+int getVNNIBlockingFactor(Type type) {
```

Comment:
Why this is not a simple constant? Why do you need to pass the type?

Reply:
We want to check the element type to be bf16/bf8. The method is a wrapper on the type check code.

Reply:
I see, here you check only for BF16. Also more than a check is an assert. Maybe return something like: 
```
Optional<int64_t> getVNNIBlockingFactor(Type t) {
if (!type.isBF16())
  return std::nullopt
return 2
}
```